tests/Fixer/AbstractFixerTest.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer;
+
+use PhpCsFixer\Tests\Fixtures\Test\AbstractFixerTest\UnconfigurableFixer;
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\AbstractFixer
+ */
+final class AbstractFixerTest extends AbstractFixerTestCase
+{
+    public function testConfigureUnconfigurable()
+    {
+        $fixer = new UnconfigurableFixer();
+
+        $this->setExpectedException('LogicException', 'Cannot configure using Abstract parent, child not implementing "PhpCsFixer\Fixer\ConfigurationDefinitionFixerInterface".');
+
+        $fixer->configure(array('foo' => 'bar'));
+    }
+
+    public function testGetConfigurationDefinitionUnconfigurable()
+    {
+        $fixer = new UnconfigurableFixer();
+
+        $this->setExpectedException('LogicException', 'Cannot get configuration definition using Abstract parent, child not implementing "PhpCsFixer\Fixer\ConfigurationDefinitionFixerInterface".');
+
+        $fixer->getConfigurationDefinition();
+    }
+
+    public function testCreateConfigurationDefinitionUnconfigurable()
+    {
+        $fixer = new UnconfigurableFixer();
+
+        $this->setExpectedException('LogicException', 'Cannot create configuration definition using Abstract parent, child not implementing "PhpCsFixer\Fixer\ConfigurationDefinitionFixerInterface".');
+
+        $fixer->doSomethingWithCreateConfigDefinition();
+    }
+
+    protected function createFixer()
```

Comment:
?

Reply:
If I don't do this, I get a `UnexpectedValueException: Cannot determine fixer class, perhaps you forget to override `getFixerName` or `createFixerFactory` method?` because it's trying to find a fixer called `abstract`. I could make this test class just extend `TestCase` instead of `AbstractFixerTestCase`?

Reply:
you don't use anything from `AbstractFixerTestCase`, no need for extending it here.